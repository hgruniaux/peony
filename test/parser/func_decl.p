//# RUN: peony %s

fn f0() {}
fn f1(i: i32) -> i32 {}
fn f2(i: i32, j: i32) {}
fn f3(i: i32, ) {}
//~^ ERROR: expected parameter declarator

fn f3() {}
//~^ ERROR: redeclaration of function 'f3'

fn f4(x: i32, x: i32) {}
//~^ ERROR: identifier 'x' is bound more than once in this parameter list

fn f5(x: i32, y: i32 = 5) {}

fn f6(x: i32 = 5, y: i32) {}
//~^ ERROR: default argument missing for parameter 2 of function 'f6'

fn f7(,) {}
//~^ ERROR: expected parameter declarator
//~| ERROR: expected parameter declarator

fn f8()
//~^ ERROR: expected function body after function declarator
