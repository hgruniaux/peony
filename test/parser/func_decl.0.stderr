<source>:6:14: error: expected parameter declarator
    6 | fn f3(i: i32, ) {}
      |              ^
<source>:9:4: error: redeclaration of function 'f3'
    9 | fn f3() {}
      |    ~~
<source>:12:15: error: identifier 'x' is bound more than once in this parameter list
   12 | fn f4(x: i32, x: i32) {}
      |               ~
<source>:17:27: error: default argument missing for parameter 2 of function 'f6'
<source>:20:7: error: expected parameter declarator
   20 | fn f7(,) {}
      |       ^
<source>:20:8: error: expected parameter declarator
   20 | fn f7(,) {}
      |        ^
<source>:24:8: error: expected function body after function declarator
   24 | fn f8()
      |        ^
<source>:27:24: error: default argument references parameter 'x'
   27 | fn f9(x: i32, y: i32 = x) {}
      |                        ~
